src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -76,7 +79,17 @@ public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeP
                     writer.Flush();
                 }
 #if NETCOREAPP
-                textWriter.Write(Encoding.UTF8.GetString(output.WrittenMemory.Span));
+                var messageBytes = output.WrittenMemory.Span;
+                var logMessageBuffer = ArrayPool<char>.Shared.Rent(Encoding.UTF8.GetMaxCharCount(messageBytes.Length));
+                try
+                {
+                    var charsWritten = Encoding.UTF8.GetChars(messageBytes, logMessageBuffer);
+                    textWriter.Write(logMessageBuffer, 0, charsWritten);
+                }
+                finally
+                {
+                    ArrayPool<char>.Shared.Return(logMessageBuffer);
+                }
 #else
                 textWriter.Write(Encoding.UTF8.GetString(output.WrittenMemory.Span.ToArray()));
```

Comment:
>output.WrittenMemory.Span.ToArray() [](http://example.com/codeflow?start=57&length=35)

This part can be optimized too by using the unsafe version of the [GetString](https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.getstring?view=net-8.0#system-text-encoding-getstring(system-byte*-system-int32)). It will save at least one allocation.